modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/client/ProcessorCache.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.runtime.client;
+
+import static com.github.benmanes.caffeine.cache.Caffeine.newBuilder;
+import static java.lang.String.format;
+
+import org.mule.runtime.extension.api.client.ExtensionsClient;
+import org.mule.runtime.extension.api.client.OperationParameters;
+import org.mule.runtime.module.extension.internal.runtime.operation.OperationMessageProcessor;
+
+import com.github.benmanes.caffeine.cache.Cache;
+
+/**
+ * Cache of {@link OperationMessageProcessor} to avoid initialization when used by the {@link ExtensionsClient}.
+ *
+ * @since 4.2.0
+ */
+public class ProcessorCache {
+
+  private static final int MAX_CACHE_SIZE = 100;
+  private static volatile ProcessorCache instance;
+
+  private static String idFormat = "Extension:%sOperation:%sConfiguration:%s";
+
+  private Cache<String, OperationMessageProcessor> cache = newBuilder().maximumSize(MAX_CACHE_SIZE).build();
+
+  private ProcessorCache() {
+    if (instance != null) {
+      throw new RuntimeException("Use getInstance() method to create and get a ProcessorCache");
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Define and throw a dedicated exception instead of using a generic one. ([squid:S00112](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS00112))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7719&issues=AWk7fTJUotIXnmsUoDsA&open=AWk7fTJUotIXnmsUoDsA)

[ISSUE_KEY]: AWk7fTJUotIXnmsUoDsA